src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -62,46 +68,32 @@ public ScheduleCalendar(com.google.api.services.calendar.Calendar client) throws
    * Puts schedule onto secondary calendar
    * 
    * @param schedule the schedule to add to the calendar 
+   * @param startDateString the user's college term start date 
+   * @param endDateString the user's college term end date
    */
-  public void addSchedule(Schedule schedule) throws IOException {
+  public void addSchedule(Schedule schedule, String startDateString, String endDateString) throws IOException {
     try {
-      // get current calendar, so can add the courses to the appropriate days of the current month 
-      java.util.Calendar calendar = java.util.Calendar.getInstance();
+      // calculate the number of weeks in their term, to set recurrence for calendar event
+      org.joda.time.LocalDate termStart = org.joda.time.LocalDate.parse(startDateString);
+      org.joda.time.LocalDate termEnd = org.joda.time.LocalDate.parse(endDateString);
+      int termLength = Weeks.weeksBetween(termStart, termEnd).getWeeks();
+
+      LocalDate startDate = LocalDate.parse(startDateString);
 
-      // for each course in the schedule, add all section times to calendar
+      // for each course in the schedule, add an event to the calendar
       for (ScheduledCourse currentCourse : schedule.getCourses()) {
-        List<TimeRange> lectureSectionTimes = currentCourse.getLectureSection().getMeetingTimes();
-        
-        addSectionToCalendar(lectureSectionTimes, calendar, currentCourse);
-        if (currentCourse.getLabSection() != null) {
-          List<TimeRange> labSectionTimes = currentCourse.getLabSection().getMeetingTimes();
-          addSectionToCalendar(labSectionTimes, calendar, currentCourse);
+        List<TimeRange> sectionTimes = currentCourse.getLectureSection().getMeetingTimes();
```

Comment:
Do you also add the `labSections` of the given course?

Reply:
I forgot to add the labSections in - thanks for seeing that!